jvav/utils.py

```diff
@@ -144,16 +168,76 @@ def get_ids_from_page(self, url: str) -> typing.Tuple[int, list]:
         try:
             soup = self.get_soup(resp)
             items = soup.find_all(class_="item")
-            ids = []
-            for item in items:
-                ids.append(item.find(class_="video-title").strong.text)
-            if ids == []:
+            ids = [item.find(class_="video-title").strong.text.strip() for item in items]
```

Comment:
no indent here, have you tested?

Reply:
* This is the **List comprehension**, a faster way to build list from a loop.
* The statement in line 171 has the same effect as the following paragraph:

  ```python
  ids = []
  for item in items:
    ids.append(item.find(class_="video-title").strong.text.strip())
  ```

* And then, it will check whether the `ids` is empty. If so, it will return 404, otherwise it will return the  `(200, <ids>)`. I think that's right.

> This is the introduction about **List comprehension**: <https://docs.python.org/3/glossary.html#term-list-comprehension>

Reply:
Ok, that's my fault, I misread the "-" before "if ids==[]" in line 150